### PR TITLE
vm deployment script along with integration and connecting GCP via API

### PR DESCRIPTION
**Pull request by SantoshDhirwani**
Added script gcp.py and integrated it into our architecture. Updated config.json.

Description
-The script connects to google cloud based on parameters set inside config.json
-The script is triggered automatically by main.py
-The script creates VMs based on the number of nodes entered by user in config.json and installs ethereum in all of them.

Pre-requisites: (Ubuntu)

sudo apt-get update && sudo apt-get install google-cloud-sdk

gcloud auth login -> opens a window/tab in your deafult browser, enter your google credentials

gcloud config set project my-project -> in our case my-project = "totemic-carrier-259013"

Related Issue
#33 

How Has This Been Tested?
Configure parameters inside config.json and then executed "python main.py" on my local machine.
However, it does not handle deleting nodes, so before executing it for the 2nd time make sure that you delete the nodes first in the GCP project.

Open for discussion:
I think this covers the requirement of #61  as well.

### CODE REVIEW

**Review by fayanpardo — CHANGES_REQUESTED** [review]
This script works but does not give the output expected.
Several things I saw (And i stopped counting or looking for more):
- The startup is not even working to install ethereum, neither geth or puppeth. There is no sudo and I guess that you need to add the needed repositories to install it.
- The machines created are not Ubuntu 18.04 which are the images we have been using since the beginning, not Debian.
- The machines created do not have HTTP enabled, so no communication is possible with them for caliper.

Basically this solution just creates a VM with a given name. Do not get me wrong I love this procedure of service accounts and Google APIs but is not even fulfilling basic things and we can not add it to the project. The only requirement to fulfill task #33 is to install ethereum with the startup script. 
Looking at this, yes, we are still going to go for task #61 as we do not have much time. You can even modify this gcp.py script to create the template. It seems that it is only half way to achieve task #61 ;)

**Comment by SantoshDhirwani** [comment]
@boxrhcp  Thank you for your comments. I worked on them and I am updating this PR. Please have a look at it again and let me know if it works as expected.

**Comment by SantoshDhirwani** [comment]
@boxrhcp  Please review the changes. Thanks!

File: bin/main.py [hunk 1]
```diff
@@ -57,14 +57,22 @@ def run_file(file_path):
...
+    intervals = range(config['test_param']['minInterval'], config['test_param']['maxInterval']+config['test_param']['intervalStep'],
         config['test_param']['intervalStep'])
-    gasLimit = range(config['test_param']['minGas'], config['test_param']['maxGas']+config['test_param']['gasStep'], 
+    gasLimit = range(config['test_param']['minGas'], config['test_param']['maxGas']+config['test_param']['gasStep'],
         config['test_param']['gasStep'])
-    
+
+    node_number = config['eth_param']['nodeNumber']
+
+    for i in range(1, node_number + 1):
```
Comment (fayanpardo) [inline bin/main.py:67]: This must not be done in the main script, if anything is done in the SUT deployment script or we loose modularity.

**Review by fayanpardo — CHANGES_REQUESTED** [review]
Nice, now the startup script is working correctly for task #33 
About the rest of implementation I tested it and I see you fixed the VM image to Ubuntu, that now is executing the startup script correctly and i can use geth and puppeth in the machines. But all this python code is still not fulfilling task #61 and the changes in main are not needed until the gcp script fulfills this task.

What I would ask you, if possible, is to add in a new branch the startup script which we know its working and make a new PR with it to merge it in master and close task #33.
The script gcp.py which is a great implementation start to deal with template creation will be merged once it fulfills #61 
Thanks!

File: bin/main.py (hunk 1 above)
Comment (SantoshDhirwani) [reply bin/main.py:67]: Yeah, you are right. I didn't think about it.

**Comment by SantoshDhirwani** [comment]
@boxrhcp  So this PR is allowing the user to connect with his/her GCP project via API call. I think it is a good feature. I know it was not a task for this sprint but I would like to have a quick chat about this tomorrow. Of course, I can move the loop (creation of an equal number of VMs as the number of nodes) outside this script.

**Comment by fayanpardo** [comment]
I agree that I like this solution more but several things that we should consider if we go further with this method:

- If we improve it, we will not use this PR, the code should not be added like that in main or we loose modularity and it needs to replicate all the SUT script logic step by step. 
- It is really dangerous to upload the service account secret and sensitive information in a public repo. We will give free access to anyone to our project basically. Another reason to not continue this PR and that should be taken care of in other branches.
- Apart from creating machines with a service account, we need to know if it  is possible to do ssh via API somehow? or at least to execute the ethereum commands. This would mean that we could get rid completely about SDK and the user has to deal with less things to run the tool
